dist/legacy-assets/js/main.js

```diff
@@ -55782,15 +55782,47 @@ function viewReleaseSelector() {
 
 function viewReportDetails(collection, isPublished, $this) {
 
-    var details, reportDetails, published, events;
+    // Enum type of filterable data
+    const filterableReportType = {
+        'dataset': 0,
+        'datasetVersion': 1
+    };
+
+    // Retrieve all information for a specific type of filterable data report
+    function getFilterableReportData(filterableDataTypeToUse, fullDatasetCollectionData) {
+        let listOfReportData = [];
+
+        // Retrieve all mandatory information fields for filterable data report
+        function getMandatoryFilterableReportInformation(datasetCollectionItem) {
+            const url = new URL(datasetCollectionItem.uri);
+            return {
+                uri: url.pathname
+            };
+        }
+
+        if (fullDatasetCollectionData != null) {
+            switch (filterableDataTypeToUse) {
```

Comment:
This is a waterfalling switch used at current, however, this enables for specific data for a filterable data type to be added, and new types to append the switch.